addons/stock/static/src/js/lazy_column_list.js

```diff
@@ -0,0 +1,160 @@
+        if (value.options && value.options.lazy) {
+            lazyColumns.push(key);
+        }
+    }
+    return lazyColumns;
+}
+
+const LazyColumnListModel = ListModel.extend({
+    /**
+     * Avoid to fetch lazy fields by filter-out for the model 
+     * (`_searchReadUngroupedList`), and let the controller manage it.
+     *
+     * @param {Object} element an element from the localData
+     * @param {Object} [options]
+     * @param {Object} [options.viewType] current viewType. If not set, we will
+     *   assume main viewType from the record
+     * @returns {string[]} the list of field names
+     */
+    _getFieldNames: function (element, options) {
+        const res = this._super.apply(this, arguments);
+        const fieldsInfo = element.fieldsInfo;
+        const viewType = (options && options.viewType) || element.viewType;
+        const lazyColumns = _getLazyColumns(fieldsInfo[viewType]);
+        return res.filter((column) => !lazyColumns.includes(column));
+    },
+
+    _makeDataPoint: function () {
+        // Should put the lazy column to work with decoration
+        const dataPoint = this._super.apply(this, arguments);
+        if (dataPoint.type === "record" && dataPoint.fieldsInfo) {
+            const fieldsInfo = dataPoint.fieldsInfo[dataPoint.viewType];
+            const lazyColumn = _getLazyColumns(fieldsInfo);
+            _bindLazyColumnToState(dataPoint, lazyColumn);
+        }
+        return dataPoint;
+    },
+});
+
+const LazyColumnListRenderer = ListRenderer.extend({
+    init: function () {
+        this.currentNbLazyRpc = 0;
+        return this._super.apply(this, arguments);
+    },
+
+    _lazyLoad: async function (state, lazyColumns) {
+        const res = await this._rpc(
+            {
+                model: state.model,
+                method: "read",
+                args: [state.res_ids, lazyColumns],
+            },
+            { shadow: true }
+        );
+        const resById = {};
+        for (const element of res.values()) {
+            resById[element.id] = {};
+            for (const lazyCol of lazyColumns) {
+                resById[element.id][lazyCol] = element[lazyCol];
+            }
+        }
+        _bindLazyColumnToState(this.state, lazyColumns, resById);
+        this.currentNbLazyRpc -= 1;
+        this._render(false);
+    },
+
+    async _render(lazy = true) {
+        if (lazy && this.state.res_ids.length > 0) {
+            // TODO don't load if the optional is hidden or all lazy column is invisible
+            this.currentNbLazyRpc += 1;
+        }
+        await this._super(...arguments);
+        if (lazy && this.state.res_ids.length > 0) {
+            const lazyColumns = _getLazyColumns(this.state.fieldsInfo[this.state.type]);
+            this._lazyLoad(this.state, lazyColumns);
+        }
+    },
+
+    _isLazyNode: function (node) {
+        try {
+            if (node.attrs && node.attrs.options && JSON.parse(node.attrs.options).lazy) {
```

Comment:
As you are in a `try/catch`, can you do directly this ? 
```suggestion
            if (JSON.parse(node.attrs.options).lazy) {
```
If `attrs` is undefined, it will run into an error and will return `false`